devito/types/sparse.py

```diff
@@ -89,38 +145,67 @@ def space_order(self):
         """The space order."""
         return self._space_order
 
+    @property
+    def r(self):
+        return self._radius
+
     @property
     def _sparse_dim(self):
         return self.dimensions[self._sparse_position]
 
     @property
-    def gridpoints(self):
-        """
-        The *reference* grid point corresponding to each sparse point.
+    def _mpitype(self):
+        return dtype_to_mpidtype(self.dtype)
 
-        Notes
-        -----
-        When using MPI, this property refers to the *physically* owned
-        sparse points.
-        """
-        raise NotImplementedError
+    @property
+    def _smpitype(self):
+        sfuncs = [getattr(self, s) for s in self._sub_functions
+                  if getattr(self, s) is not None]
+        return {s: dtype_to_mpidtype(s.dtype) for s in sfuncs}
 
-    def interpolate(self, *args, **kwargs):
-        """
-        Implement an interpolation operation from the grid onto the given sparse points
-        """
-        return self.interpolator.interpolate(*args, **kwargs)
+    @property
+    def _comm(self):
+        return self._distributor.comm
 
-    def inject(self, *args, **kwargs):
-        """
-        Implement an injection operation from a sparse point onto the grid
-        """
-        return self.interpolator.inject(*args, **kwargs)
+    @property
+    def _coords_indices(self):
+        if self.gridpoints_data is not None:
+            return self.gridpoints_data
+        else:
```

Comment:
no need for this else

Reply:
THough you said we prefer if/else rather than just if and then more code but can change it

Reply:
No no I meant something completely different... but not sure now what my brain was thinking. Ignore me, resolving